tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+
+        unique_marker = str(uuid.uuid4())
+
+        snapshot.add_transformer(snapshot.transform.regex(unique_marker, "<unique-marker>"))
+
+        detail = {"i": 0, "marker": unique_marker}
+
+        aws_client.events.put_targets(
+            Rule=rule_name,
+            Targets=[
+                {
+                    "Id": target_id,
+                    "Arn": api_destination_arn,
+                    "Input": json.dumps({"target_value": "value", "marker": unique_marker}),
+                    "RoleArn": role_arn,
+                    "HttpParameters": {
+                        "HeaderParameters": {"target-header": "target_header_value"},
+                        "QueryStringParameters": {"target_query": "t_query"},
+                    },
+                }
+            ],
+        )
+
+        try:
+            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{backend_function_name}")
+        except aws_client.logs.exceptions.ResourceAlreadyExistsException:
+            pass
+
+        try:
+            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{oauth_function_name}")
+        except aws_client.logs.exceptions.ResourceAlreadyExistsException:
+            pass
+
+        aws_client.events.put_events(
+            Entries=[
+                {"Source": "source-123", "DetailType": "type-123", "Detail": json.dumps(detail)}
+            ]
+        )
+
+        def _check_backend_logs(retries=0):
+            events = get_lambda_log_events(
+                function_name=backend_function_name, logs_client=aws_client.logs
+            )
+            filtered = []
+            for e in events:
+                if isinstance(e, dict) and "body" in e:
+                    try:
+                        body = json.loads(e["body"])
+                        if body.get("marker") == unique_marker:
+                            filtered.append(e)
+                    except Exception as ex:
+                        print("Exception parsing event body:", ex)
+            if len(filtered) == 1:
+                return filtered[0]
+            raise AssertionError(
+                f"Expected exactly 1 invocation with marker {unique_marker}, found {len(filtered)}"
+            )
+
+        event_log = retry(_check_backend_logs, retries=30, sleep=5)
+        headers = {k.lower(): v for k, v in event_log.get("headers", {}).items()}
+        posted_body = json.loads(event_log.get("body", "{}"))
+        qs_params = event_log.get("queryStringParameters", {}) or {}
+
+        snapshot.match(
+            "final_invocation", {"headers": headers, "body": posted_body, "qs_params": qs_params}
+        )
+
+        assert posted_body["target_value"] == "value"
+        assert posted_body["marker"] == unique_marker
+        assert headers.get("target-header", "").lower() == "target_header_value"
+        assert qs_params.get("target_query") == "t_query"
+        assert qs_params.get("overwritten_query") == "original"
```

Comment:
question: do we need those asserts? They should be covered by the snapshot, right?